### PR TITLE
Remove tags and branch heads fetching

### PR DESCRIPTION
**Pull request by atuchin-m**
Resolves 

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: build/commands/lib/util.js [hunk 1]
```diff
@@ -638,7 +638,7 @@ const util = {
     // base args
     const initialArgs = ['sync', '--reset', '--nohooks']
     const chromiumArgs = ['--revision', 'src@' + config.getProjectRef('chrome')]
-    const resetArgs = ['--with_tags', '--with_branch_heads', '--upstream']
```
Comment (petemill) [inline build/commands/lib/util.js:641]: don't we refer to chromium versions as tags and won't this remove fetching them?

Comment (atuchin-m) [reply build/commands/lib/util.js:641]: gclient has an ability to fetch unknown commit . But let me check twice;)

Comment (atuchin-m) [reply build/commands/lib/util.js:641]: But this PR builds even with `TERMINATE_NODE` which should wipe gitcache

File: build/commands/lib/util.js [hunk 2]
```diff
@@ -638,7 +638,7 @@ const util = {
...
+    const resetArgs = ['--upstream']
```
Comment (atuchin-m) [inline build/commands/lib/util.js:641]:
```suggestion
    const resetArgs = ['--with_tags', '--upstream']
```

File: build/commands/lib/util.js (hunk 1 above)
Comment (atuchin-m) [reply build/commands/lib/util.js:641]: Unfortunately I'm not sure that removing tags will not break checkout in some corner cases. gclient inside has a obscure behaviour with a lot of cases.
But `with_branch_heads` can be safely removed, lets do it step by step.

Comment (mherrmann) [reply build/commands/lib/util.js:649]: @bridiver "accidentally" removed `with_branch_heads` in April 2020 then added it back 11 days later ([associated commit](https://github.com/brave/brave-core/commit/cd2bef907dd5e3aca2e09c5d64615e8eafc65bef)). I couldn't find a discussion of the flag in the Slack channels I have access to, nor here on GitHub. But the fact that it was added back 11 days after being removed makes me think there might be a reason why it's there?

Comment (atuchin-m) [reply build/commands/lib/util.js:649]: I have to agree that it is simple to leave with_branch_heads here than get a build issue in the future.
I've found that 3rd parties also checkouts with this flag and have these branches.

So, let's decline this PR

**Pull request closed without merging**